bin/ch/WScriptJsrt.cpp

```diff
@@ -1490,7 +1527,22 @@ HRESULT WScriptJsrt::ModuleMessage::Call(LPCSTR fileName)
         errorCode = specifierStr.GetError();
         if (errorCode == JsNoError)
         {
+            char oldpath[_MAX_PATH];
+            bool useReferenceModulePath = false;
+            if (this->moduleRecord)
+            {
+                auto moduleDirEntry = moduleDirMap.find(this->moduleRecord);
+                if (moduleDirEntry != moduleDirMap.end())
+                {
+                    _getcwd(oldpath, _MAX_PATH);
+                    useReferenceModulePath = true;
+                    std::string dir = moduleDirEntry->second;
+                    _chdir(dir.c_str());
```

Comment:
Chdir happens in FetchImportedModule*() or ModuleMessage::Call(). I thought only the main thread will go to these two places. No?

---
In reply to: [125371146](https://github.com/Microsoft/ChakraCore/pull/3284#discussion_r125371146) [](ancestors = 125371146)